Client/src/python/dbs/apis/dbsClient.py

```diff
@@ -1015,13 +1057,19 @@ def listFileArray(self, **kwargs):
             for slice in slicedIterator(sourcelist, max_list_len):
                 kwargs[mykey] = slice
                 results.extend(self.__callServer("fileArray", data=kwargs, callmethod="POST"))
-            #make sure only one dictionary per lfn.
-	    #Make sure this changes when we move to 2.7 or 3.0
-	    #http://stackoverflow.com/questions/11092511/python-list-of-unique-dictionaries
-            # YG May-26-2015	
-            return dict((v['logical_file_name'],v) for v in results).values()
+        elif split_lumi_list:
+            for item in split_lumi_list:
+                kwargs['lumi_list'] = item
+                results.extend(self.__callServer("fileArray", data=kwargs, callmethod="POST"))
         else:
             return self.__callServer("fileArray", data=kwargs, callmethod="POST")
+        
+        #make sure only one dictionary per lfn.
+        #Make sure this changes when we move to 2.7 or 3.0
+        #http://stackoverflow.com/questions/11092511/python-list-of-unique-dictionaries
+        # YG May-26-2015
+        return dict((v['logical_file_name'],v) for v in results).values()
```

Comment:
I'm sure the old form still works in 2.7, but yeah, 2.7 has the dictionary comprehension.